### PR TITLE
Add preview action

### PR DESCRIPTION
**Pull request by shiena**
Shows the information in the preview window.

### CODE REVIEW

File: autoload/unite/sources/hoogle.vim [hunk 1]
```diff
@@ -43,6 +45,7 @@ function! s:parse(input, line)
   else
     let l:candidate.action__haddock_module = l:mod
     let l:candidate.action__haddock_fragment = matchstr(l:rest, '#.\+$')
+    let l:candidate.action__haddock_index = a:index
```
Comment (eagletmt) [inline autoload/unite/sources/hoogle.vim]: Why `action__haddock_index` isn't set when `l:mod == 'package'`?

Comment (shiena) [inline autoload/unite/sources/hoogle.vim]: It was my mistake.
I added a fixed in shiena/unite-haddock@a57777d22adad093a487233c97dac33f3f67cb17

**Comment by eagletmt** [comment]
E108 occurs in the following scenario.
I'm using https://github.com/Shougo/unite.vim/commit/8274e17999dbd277ace0622865c33dc31b9aa2c2 .
1. Execute `:Unite hoogle` and enter some keywords.
2. Move to a candidate, and press `<Tab>` (`<Plug>(unite_choose_action)`)
3. Choose `preview` action and the preview window is opened.
4. Press `q` (`<Plug>(unite_exit)`) to close unite buffer.
5. Quit Vim, and `Vim(unlet):E108: No such variable: "s:exact"` occurs.

It seems that `on_close` hook is called twice at step 4 and 5.
I don't know it's an intended behavior or a bug of unite.vim.
What do you think?

**Comment by eagletmt** [comment]
In general, storing a local state to a global variable is a bad idea.
How about saving `--exact` flag to a candidate?

**Comment by shiena** [comment]
> It seems that on_close hook is called twice at step 4 and 5.
> I don't know it's an intended behavior or a bug of unite.vim.
> What do you think?

I have reproduced here as well.
I was thinking of the exact clear in on_close for just in case.
I do not know why the on_close is hook twice.
But, I think it safe to delete the on_close.

> In general, storing a local state to a global variable is a bad idea.
> How about saving --exact flag to a candidate?

I see. I also think the how is good.

**Comment by eagletmt** [comment]
Okay, then I merge this PR.
Thank you for adding a cool feature.

**Comment by shiena** [comment]
Thank you incorporate my patch!

**Pull request merged**